src/helpers/util.js

```diff
@@ -39,5 +39,32 @@ module.exports = {
       ret.push(row)
     }
     return ret
+  },
+
+  /*
+    binary search
```

Comment:
This deserves a better explanation, also `changes` should be renamed. Is the following correct?

Assumes that `changes` is sorted increasingly and returns the smallest `i` such that `target <= changes[i]` (or `changes.length` if all elements in `changes` are smaller than `target`).